cli/tests/test_templater.rs

```diff
@@ -356,6 +374,32 @@ fn test_templater_string_method() {
 
     insta::assert_snapshot!(render(r#""".lines()"#), @"");
     insta::assert_snapshot!(render(r#""a\nb\nc\n".lines()"#), @"a b c");
+
+    insta::assert_snapshot!(render(r#""".starts_with("")"#), @"true");
+    insta::assert_snapshot!(render(r#""everything".starts_with("")"#), @"true");
+    insta::assert_snapshot!(render(r#""".starts_with("foo")"#), @"false");
+    insta::assert_snapshot!(render(r#""foo".starts_with("foo")"#), @"true");
+    insta::assert_snapshot!(render(r#""foobar".starts_with("foo")"#), @"true");
+    insta::assert_snapshot!(render(r#""foobar".starts_with("bar")"#), @"false");
+
+    insta::assert_snapshot!(render(r#""".ends_with("")"#), @"true");
+    insta::assert_snapshot!(render(r#""everything".ends_with("")"#), @"true");
+    insta::assert_snapshot!(render(r#""".ends_with("foo")"#), @"false");
+    insta::assert_snapshot!(render(r#""foo".ends_with("foo")"#), @"true");
+    insta::assert_snapshot!(render(r#""foobar".ends_with("foo")"#), @"false");
+    insta::assert_snapshot!(render(r#""foobar".ends_with("bar")"#), @"true");
+
+    insta::assert_snapshot!(render(r#""".remove_prefix("wip: ")"#), @"");
+    insta::assert_snapshot!(render(r#""wip: testing".remove_prefix("wip: ")"#), @"testing");
+
+    insta::assert_snapshot!(render(r#""bar@my.example.com".remove_suffix("@other.example.com")"#), @"bar@my.example.com");
+    insta::assert_snapshot!(render(r#""bar@other.example.com".remove_suffix("@other.example.com")"#), @"bar");
+
+    insta::assert_snapshot!(render(r#""foo".substr(0, 0)"#), @"");
+    insta::assert_snapshot!(render(r#""foo".substr(0, 1)"#), @"f");
+    insta::assert_snapshot!(render(r#""foo".substr(0, 99)"#), @"foo");
+    insta::assert_snapshot!(render(r#""abcdef".substr(2, -1)"#), @"cde");
+    insta::assert_snapshot!(render(r#""abcdef".substr(-3, 99)"#), @"def");
```

Comment:
`"abcdef".substr(4, 2)` and/or `"abcdef".substr(-2, -4)` might also be useful to test (it looks like it's handled correctly in the code, but would be good to exercise that code path)